File: Ahorn/entities/DieOnFlagsController.jl
module AurorasHelperDieOnFlagsController

using ..Ahorn, Maple

@mapdef Entity "AurorasHelper/DieOnFlagsController" DieOnFlagsController(x::Integer, y::Integer, Flags::String="death_flags", FlagsMinimumFrames::String="30", LeniencyFrames::Integer=15)

const placements = Ahorn.PlacementDict(
    "Die on Flags Controller (Aurora's Helper)" => Ahorn.EntityPlacement(
        DieOnFlagsController,
        "rectangle"
    )
)

function Ahorn.selection(entity::DieOnFlagsController)
    x, y = Ahorn.position(entity)

    return Ahorn.Rectangle(x - 12, y - 12, 24, 24)
end


Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::DieOnFlagsController, room::Maple.Room) = Ahorn.drawSprite(ctx, "controllers/AurorasHelper/DieOnFlagsController", 0, 0)


end

File: Loenn/entities/DieOnFlagsController.lua
local DieOnFlagsController = {}

DieOnFlagsController.name = "AurorasHelper/DieOnFlagsController"
DieOnFlagsController.depth = 0
DieOnFlagsController.texture = "controllers/AurorasHelper/DieOnFlagsController"
DieOnFlagsController.justification = {0.5, 1.0}

DieOnFlagsController.placements = {
    name = "DieOnFlagsController",
    data = {
        Flags = "timed_flag",
        FlagsMinimumFrames = "30",
        LeniencyFrames = 15,
    }
}

DieOnFlagsController.fieldInformation = {
    Flags = {
        fieldType = "string"
    },
    FlagsMinimumFrames = {
        fieldType = "string"
    },
    LeniencyFrames = {
        fieldType = "integer"
    },
}

return DieOnFlagsController